coderd/notifications/dispatch/smtp_test.go

```diff
@@ -381,6 +382,21 @@ func TestSMTP(t *testing.T) {
 			toAddrs:          []string{to},
 			expectedAuthMeth: sasl.Plain,
 		},
+		/**
+		 * Other errors
+		 */
+		{
+			name: "Rejected on DATA",
+			cfg: codersdk.NotificationsEmailConfig{
+				Hello: hello,
+				From:  from,
+			},
+			failOnDataFn: func() error {
+				return &smtp.SMTPError{Code: 501, EnhancedCode: smtp.EnhancedCode{5, 5, 4}, Message: "Rejected!"}
+			},
+			expectedErr: "SMTP error 501: Rejected!",
+			retryable:   true,
```

Comment:
Isn't this error non-retryable?

Reply:
Not apriori; it could be a temporary failure.